### PR TITLE
ui: allow for regular expressions to be used in search (PROJQUAY-6597)

### PR DESCRIPTION
**Pull request by dmesser**
This adds filtering capabilities based on regular expressions to the org list, repo list and tag list views. This is useful for more complex filtering of vast amounts of org/repos/tags or for validating a regular expression for mirroring configs in the UI or for custom automation built by users.

This is enabled by using PatternFly's advanced search menu from the native `SearchInput` component and could set the precedence for even more complex search scenarios once we support them.
The search field is also a little wider now for convenience when entering a more complex pattern. This ate up more space in the toolbar, especially when the checkbox dropdown "select all/select page/select nothing" would also say how many tags were selected, e.g., "20 selected" and the "Actions" menu button appeared. To save space and prevent line breaks in the toolbar, I have replaced that with a little badge that still complies with PatternFly style guides.

The PR also cleans up the org list and tag list view by removing the dropdown, which currently has a single entry only ("Name"). We can easily add it back when needed.

The dropdown fields for the tag list view search were renamed to "Tag" and "Digest" for clarity (previously "Name" and "Manifest").

The implementation currently doesn't hide the regex search checkbox when the user selects 

The regular expression is purely evaluated on the client side, so [ReDoS](https://en.wikipedia.org/wiki/ReDoS) server attacks are impossible. But it also depends on all the tags being fetched, so we need to teach future APIs how to do this server-side sooner or later.

Screenshot 2024-01-15 at 22 51 57
Screenshot 2024-01-15 at 22 51 48
Screenshot 2024-01-15 at 22 51 26
Screenshot 2024-01-15 at 22 49 50
Screenshot 2024-01-15 at 22 49 26
Screenshot 2024-01-15 at 22 49 20
Screenshot 2024-01-15 at 22 49 10

### CODE REVIEW

**Review by harishsurf — COMMENTED** [review]
Overall looks good, few comments: 
1. The text entered in the search box is not cleared when switching orgs/ repos.
2. In tags page, the regex doesn't seem to apply when searching based on `Digest`. I'm thinking if we want to have regex at all for a digest since they follow the same format?
3. super minor and can be skipped for now - for some reason the regex dropdown needs to clicked twice for it to reopen again

File: web/src/components/toolbar/FilterInput.tsx [hunk 1]
```diff
@@ -1,26 +1,113 @@
...
+      id="filter-input-advanced-search"
+    >
+      <PanelMain>
+        <PanelMainBody>
+          <Checkbox
+            label="Use regular expressions"
+            id="filter-input-regex-checker"
+            name="filter-input-regex-checkbox"
+            isChecked={props.searchState.isRegEx}
+            onChange={(e, checked) =>
```
[the remark this answers is not in the document]
Comment (dmesser) [reply web/src/components/toolbar/FilterInput.tsx:33]: When I tried that the click event was processed so quickly that the user can't actually see if the checkbox reacted to click by displaying the checkmark. I would say that is confusing UX.

File: web/src/routes/RepositoriesList/RepositoriesList.tsx [hunk 2]
```diff
@@ -93,15 +94,14 @@ export default function RepositoriesList(props: RepositoriesListProps) {
     } as RepoListTableItem;
   });
 
+  if (currentOrg !== null && search.currentOrganization !== currentOrg) {
+    setSearch({...search, currentOrganization: currentOrg});
```
[the remark this answers is not in the document]
Comment (dmesser) [reply web/src/routes/RepositoriesList/RepositoriesList.tsx]: Should be fixed now.

**Comment by dmesser** [comment]
Thank you for the review!
> 
> 1. The text entered in the search box is not cleared when switching orgs/ repos.

Great find, should be fixed now.

> 2. In tags page, the regex doesn't seem to apply when searching based on `Digest`. I'm thinking if we want to have regex at all for a digest since they follow the same format?

Initially I thought doing regex search for digest queries has no relevance. But since I can't not show the advanced search dropdown based on the selection of digest as a search criteria, I added support for regular expressions here as well.

> 3. super minor and can be skipped for now - for some reason the regex dropdown needs to clicked twice for it to reopen again

Should be fixed with the recent commits.

**Review by harishsurf — APPROVED** [review]
lgtm!

**Pull request merged**